### ISSUE TITLE
[web] built web project errors

### ISSUE THREAD

**Issue by chrisdrackett** [expo/expo-cli#614]
I'm having an issue with a web project. Running locally I have no problems, but after running build and deploying to now as a static site I'm getting the following in my console:

```
Uncaught TypeError: Cannot read property 'call' of undefined
    at f (runtime~app.a8a9905a.js:1)
    at Object.<anonymous> (2.2c12fd3d.chunk.js:1)
    at f (runtime~app.a8a9905a.js:1)
    at Module.453 (app.0f9f7f5c.chunk.js:1)
    at f (runtime~app.a8a9905a.js:1)
    at Object.212 (app.0f9f7f5c.chunk.js:1)
    at f (runtime~app.a8a9905a.js:1)
    at t (runtime~app.a8a9905a.js:1)
    at Array.r [as push] (runtime~app.a8a9905a.js:1)
    at app.0f9f7f5c.chunk.js:1
```

I'm running:

*expo-cli*: 2.19.4

steps to reproduce:

1. init project via expo:
```
> expo --version
2.19.4
> expo init expo
```

2. chose `blank`
3. yarn found, install with yarn (yes)
4. `cd expo && expo build:web`
5. deploy web-build to a static host (I used now)
6. result: https://web-build-foskvwojae.now.sh

```
runtime~app.2dbaae62.js:1 Uncaught TypeError: Cannot read property 'call' of undefined
    at f (runtime~app.2dbaae62.js:1)
    at Object.<anonymous> (2.95e6a84b.chunk.js:1)
    at f (runtime~app.2dbaae62.js:1)
    at Module.<anonymous> (2.95e6a84b.chunk.js:1)
    at f (runtime~app.2dbaae62.js:1)
    at Object.135 (app.9a2fc698.chunk.js:1)
    at f (runtime~app.2dbaae62.js:1)
    at t (runtime~app.2dbaae62.js:1)
    at Array.r [as push] (runtime~app.2dbaae62.js:1)
    at app.9a2fc698.chunk.js:1
f @ runtime~app.2dbaae62.js:1
(anonymous) @ 2.95e6a84b.chunk.js:1
f @ runtime~app.2dbaae62.js:1
(anonymous) @ 2.95e6a84b.chunk.js:1
f @ runtime~app.2dbaae62.js:1
135 @ app.9a2fc698.chunk.js:1
f @ runtime~app.2dbaae62.js:1
t @ runtime~app.2dbaae62.js:1
r @ runtime~app.2dbaae62.js:1
(anonymous) @ app.9a2fc698.chunk.js:1
```

**Comment by chrisdrackett** [comment]
I've spent some time to make a minimal test case here:

https://github.com/chrisdrackett/expo-mono-test

within this repo if you run:

- `yarn`

then within the `expo` folder `yarn build`

you'll end up with a `build` folder that gives the following result: https://build-vajyuibada.now.sh

**Comment by chrisdrackett** [comment]
I wonder if this is related to https://github.com/webpack/webpack/issues/8911

**Comment by chrisdrackett** [comment]
I was able to further simplify this example. I've removed:

- yarn workspaces
- typescript

new broken project here at the same place, just updated master: https://github.com/chrisdrackett/expo-mono-test

**Comment by chrisdrackett** [comment]
note this project does build with CRA, not sure if that helps or not

**Comment by chrisdrackett** [comment]
I've updated the test project to expo 33 (I believe) and I'm still seeing the same behavior. Test project is still here:

https://github.com/chrisdrackett/expo-mono-test

/cc @EvanBacon

**Comment by chrisdrackett** [comment]
I decided to create a project using the expo-cli and was able to get the same results with a build. I put this work in a branch here: https://github.com/chrisdrackett/expo-mono-test/tree/using-expo-starter

**Comment by chrisdrackett** [comment]
1. init project via expo:

```
> expo --version
2.19.4
> expo init expo
```

2. chose `blank`
3. yarn found, install with yarn (yes)
4. `cd expo && expo build:web`
5. deploy `web-build` to a static host (I used now)
6. result: https://web-build-foskvwojae.now.sh

```
runtime~app.2dbaae62.js:1 Uncaught TypeError: Cannot read property 'call' of undefined
    at f (runtime~app.2dbaae62.js:1)
    at Object.<anonymous> (2.95e6a84b.chunk.js:1)
    at f (runtime~app.2dbaae62.js:1)
    at Module.<anonymous> (2.95e6a84b.chunk.js:1)
    at f (runtime~app.2dbaae62.js:1)
    at Object.135 (app.9a2fc698.chunk.js:1)
    at f (runtime~app.2dbaae62.js:1)
    at t (runtime~app.2dbaae62.js:1)
    at Array.r [as push] (runtime~app.2dbaae62.js:1)
    at app.9a2fc698.chunk.js:1
f @ runtime~app.2dbaae62.js:1
(anonymous) @ 2.95e6a84b.chunk.js:1
f @ runtime~app.2dbaae62.js:1
(anonymous) @ 2.95e6a84b.chunk.js:1
f @ runtime~app.2dbaae62.js:1
135 @ app.9a2fc698.chunk.js:1
f @ runtime~app.2dbaae62.js:1
t @ runtime~app.2dbaae62.js:1
r @ runtime~app.2dbaae62.js:1
(anonymous) @ app.9a2fc698.chunk.js:1
```

**Comment by raajnadar** [comment]
I think you need a break check this repo https://github.com/expo/web-examples/tree/master/create-react-app-example

**Comment by chrisdrackett** [comment]
a break? I'm just trying to report an issue I'm seeing. At this point I don't think this has anything to do with CRA as I'm able to reproduce with just the default unedited output from an `expo init` project. It is true that the particular example you link works, but as you start to add components it'll break down as well.

**Comment by EvanBacon** [comment]
We have a solution merged which will disable tree shaking so arbitrary components which haven't been optimized properly can be run without errors being thown. Unfortunately there are issues with NPM preventing me from publishing this fix. https://github.com/expo/expo/pull/4151

**Comment by EvanBacon** [comment]
When running in prod the tree-shaking removes any code it deems "unused". An example of this would be:
```js
import "./index.css"
```
To let the bundler know that this code is in use, you can define the file as a "sideEffect" in the `package.json`. Here's an example of a working configuration:
```js
{
  "sideEffects": [
    "src/*",
    "!src/components/*"
  ],
}
```

**Comment by ruslan-mystore** [comment]
same problem here

**Comment by aarshaw** [comment]
I'm also getting this issue

**Comment by chrisdrackett** [comment]
no, this is still an issue in expo `2.20.2`. I am running the test directly on an `expo init` project as in `2.20.2` I can't even build my own app via. expo, I get the following output:

```
$ expo build:web
Warning: This project contains unoptimized assets. Please run `expo optimize` in your project directory.
✨  Done in 7.10s.
```

that being said, I plan on getting the `expo init` case working before I open an issue for my own app.

@EvanBacon did your fix make it into a recent release? If so this issue might need to be reopened.

**Comment by jaimereyley** [comment]
same problem, I'm using expo `2.20.2` and `expo init`. Maybe [webpack/webpack#8996](https://github.com/webpack/webpack/issues/8996) is related.

**Comment by aarshaw** [comment]
Was able to workaround using the following in my webpack config:

```
const _config = require('@expo/webpack-config')

module.exports = async (env, argv) => {
  const config = await _config(env, argv)
  if (config.mode === 'production') {
    config.mode = 'none'
  }
  return config
}
```

**Comment by jaimereyley** [comment]
@aarshaw great! Your solution worked for me!

**Comment by chrisdrackett** [comment]
@aarshaw I haven't tried yet, but I'm guessing this just runs an unoptimized (dev?) build in production?

**Comment by chrisdrackett** [comment]
@EvanBacon @cruzach I'm not sure the best course of action on this issue. I am still seeing the same behavior on the most recent release of expo-cli (that works, 2.20.3 won't start at all due to #755).

I don't want to open a new issue as I'm worried it will be seen as a duplicate, but this issue is closed and my experience tells me that closed issues don't get attention.

**Comment by Nemesisesq** [comment]
I'm curious as to why this is getting no love.

**Comment by Nemesisesq** [comment]
@aarshaw solution worked!!!

**Comment by chrisdrackett** [comment]
I'm not a fan of duplicate issues but I went ahead and opened a new issue here: https://github.com/expo/expo-cli/issues/778 as I'm not able to re-open this issue.

**Comment by EvanBacon** [comment]
investigating this now

**Comment by EvanBacon** [comment]
> Deep scope analysis was rewritten and only the patch version was bumped. I suspect this could be what broke web builds.

Related: https://github.com/expo/expo-cli/pull/761 (Not merged as of `@expo/webpack-config@0.5.18`)

**Comment by bidah** [comment]
> > Deep scope analysis was rewritten and only the patch version was bumped. I suspect this could be what broke web builds.
> 
> Related: #761 (Not merged as of `@expo/webpack-config@0.5.18`)

downloaded expo cli v2.21.2 and error is still there
